metrics/config.go

```diff
@@ -134,28 +167,30 @@ func UpdateExporterFromConfigMap(domain string, component string, logger *zap.Su
 			return
 		}
 
-		if isMetricsConfigChanged(newConfig) {
-			if err := newMetricsExporter(newConfig, logger); err != nil {
+		if isNewExporterRequired(newConfig) {
```

Comment:
Should this function return an error?

Reply:
I don't see any case it could return an error.

Reply:
well you log an error. If it's not an error, then logging it as an error, also doesn't make sense?

Reply:
The error is returned from `newMetricsExporter`. I think you were asking whether `isNewExporterRequired` should return an error?